src/mlpack/methods/decision_tree/all_dimension_select.hpp

```diff
@@ -25,35 +25,24 @@ class AllDimensionSelect
   /**
    * Construct the AllDimensionSelect object for the given number of dimensions.
    */
-  AllDimensionSelect() : i(0), dimensions(0) { }
+  AllDimensionSelect() : dimensions(0) { }
 
   /**
-   * Get the first dimension to select from.
+   * Get the current dimension.
```

Comment:
```suggestion
   * Get the `i`th dimension.
```
Just a little pedantic comment fix. :)